bcgov/biohubbc-platform
# BHBC-2228: Refactor API For Frontend Calls - Spatial

# Overview

## Links to Jira tickets

- https://quartech.atlassian.net/browse/BHBC-2228

## Description of relevant changes

- updated SQL statements to reflect new table structures
- changes to functions that need to use the submission_observation_id instead of the submission_id
- updated migrations and seeds
- parse spatial records for taxonId instead of associatedTaxa

Easiest way to test is to send these two zip files to Biohub through Postman.  They are from the same project, but two different surveys (one for Moose and one for Sheep).  Sheep is a protected species, and should only be seen by admins or by someone who has an exception to the security rule:

[cd1161ef-667c-4074-a041-6656eb94fa85.zip](https://github.com/bcgov/biohubbc-platform/files/10913677/cd1161ef-667c-4074-a041-6656eb94fa85.zip)
[846d6db4-b6f3-4a79-a32f-05bfe2b6e403.zip](https://github.com/bcgov/biohubbc-platform/files/10913679/846d6db4-b6f3-4a79-a32f-05bfe2b6e403.zip)

To test user exeptions:
* log in with a bceid account
* the user should be able to see moose records but not sheep records
* insert a user exception record for sheep for that user
* and the user should now be able to see both sheep and moose records
